crates/ruff_python_ast/src/lib.rs

```diff
@@ -86,11 +86,15 @@ impl PySourceType {
     ///
     /// Falls back to `Python` if the extension is not recognized.
     pub fn from_extension(extension: &str) -> Self {
+        Self::try_from_extension(extension).unwrap_or_default()
+    }
+
+    pub fn try_from_extension(extension: &str) -> Option<Self> {
         match extension {
-            "py" => Self::Python,
-            "pyi" => Self::Stub,
-            "ipynb" => Self::Ipynb,
-            _ => Self::Python,
+            "py" => Some(Self::Python),
+            "pyi" => Some(Self::Stub),
+            "ipynb" => Some(Self::Ipynb),
+            _ => None,
         }
     }
 }
```

Comment:
I thought the decision in https://github.com/astral-sh/ruff/issues/12250 was that the current behaviour here was correct

Reply:
I'm not changing the existing behavior. I just need an easy way for now to filter out files that we don't understand without relying on the configuration (which defaults to python files only)

Reply:
Oh I see, you've added an extra method. Sorry!

Reply:
No worries. Would have been bad if I did change the behavior 😂